src/operator/nn/mkldnn/mkldnn_lrn-inl.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file mkldnn_lrn-inl.h 
+ * \brief
+ * \Author: Patric Zhao, patric.zhao@intel.com
+*/
+#ifndef MXNET_OPERATOR_NN_MKLDNN_MKLDNN_LRN_INL_H_
+#define MXNET_OPERATOR_NN_MKLDNN_MKLDNN_LRN_INL_H_
+
+#if MXNET_USE_MKLDNN == 1
+#include <mkldnn.hpp>
+#include "../lrn-inl.h"
+#include "./mkldnn_base-inl.h"
+
+namespace mxnet {
+namespace op {
+
+static inline algorithm GetMKLDNNLRNAlgo(const LRNParam &param) {
+  // TODO(Patric): lrn_within_channel will cause core dump in MKLDNN backward
+  //               Need to confirm with MKLDNN team and fix later
+  return algorithm::lrn_across_channels;
+}
+
+inline static lrn_forward::primitive_desc GetLRNFwd(
+    const LRNParam &param, bool is_train, const memory::desc &src_md) {
+  auto engine = CpuEngine::Get()->get_engine();
+  auto alg_ = GetMKLDNNLRNAlgo(param);
```

Comment:
Coding style suggestions:
- Local variables do not normally have underscores appended. It's used for data members of classes.
- For simple data types, suggest not use `auto`. It results in poor code readability.
- Add `const` qualifier for the local variables if you can.

Reply:
Thanks for the comments. Will update the code based on your suggestions.

Reply:
@reminisce We will try to remove `auto` for simple data types. But still keep some for mkldnn types, since it will be too long when I expand the whole namespace. I will add comments before those `auto` keyword to improve code readability.